lib/src/model/model_node.dart

```diff
@@ -47,11 +45,25 @@ class ModelNode {
 
   String get sourceCode {
     if (_sourceCode == null) {
-      if (_sourceOffset != null) {
+      if (_sourceNode?.offset != null) {
+        var sourceNode = _sourceNode;
```

Comment:
optional: might be less confusing to name `sourceNode` something else

Reply:
Sure! I'm thinking of renaming it to `astNode`. Would that be fine?

Reply:
Maybe 'enclosingSourceNode' or something along those lines, since it is based on `_sourceNode` and may be computed based on parent nodes?

Reply:
Done!